crypto/pem/pem_lib.c

```diff
@@ -679,10 +679,33 @@ int PEM_read(FILE *fp, char **name, char **header, unsigned char **data,
 }
 #endif
 
+typedef struct {
+    int bom_len;
+    char bom[4];
+} bom_st;
+
+bom_st boms[5] = {
+    {3, {0xEF, 0xBB, 0xBF, 0x00} }, /* UTF-8 */
+    {2, {0xFE, 0xFF, 0x00, 0x00} }, /* UTF-16be */
+    {2, {0xFF, 0xFE, 0x00, 0x00} }, /* UTF-16le */
+    {4, {0x00, 0x00, 0xFE, 0xFF} }, /* UTF-32be */
+    {4, {0xFF, 0xFE, 0x00, 0x00} }, /* UTF-32le */
```

Comment:
You can't just ignore these diverse BOMs. If they are present, they must have meaning, and if we face a utf-16 or utf-32 file and pretend to handle it, we *must* read them as 16- or 32-bit chars rather than 8-bit.  But, we don't claim to handle anything but US-ASCII, ISO-8859-1 or UTF-8, so the sane thing would be that for all BOMs other than utf-8, we should just err.

Reply:
Got it and agreed. Will fix tomorrow.

Reply:
Wouldn't it have been better to keep the tests for the other UTF formats and give a meaningful error message here (e.g. "UTF-16 encoding not supported"), instead of just continuing and wait for some obscure BIO error message?

Reply:
Not sure. I've never seen such files in practice, but see the UTF8 ones more or less frequently.

I've also seen the messenger converting -- to – causing applications crash on cut'n'paste...